app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -38,8 +38,14 @@ type Opts struct {
 	Stderr    io.Writer
 }
 
-func New(opts Opts) *Envoy {
-	return &Envoy{opts: opts}
+func New(opts Opts) (*Envoy, error) {
+	binaryPathConfig := opts.Config.DataplaneRuntime.BinaryPath
+	_, err := lookupEnvoyPath(binaryPathConfig)
+	if err != nil {
```

Comment:
[minor]: I'd prefer if this one be one-lined:
```go
if _, err := lookupEnvoyPath(opts.Config.DataplaneRuntime.BinaryPath); err != nil {
    runLog.Error(err, "could not find the envoy executable in your path")
    return nil, err
}

Reply:
changed